.github/coding-style.md

```diff
@@ -0,0 +1,171 @@
+fall into what can be considered common Go style, please add it to this guide.
+Thanks!
+
+## On consistency
+
+There are often many ways to express the same thing (and not all of them are
+correct). Consistency in code serves two main purposes:
+
+- Avoiding errors of ignorance.
+- Minimizing reader's surprise and hence facilitate reviews and maintenance.
+
+When you deviate from a common pattern, consider leaving a comment explaining
+why you do so. Prefer local consistency over global if you have to choose.
+
+## On readability
+
+One of the things that significantly impairs code readability is the amount of
+non-local reasoning required to understand the contract and assumptions. Strive
+to design your types and interfaces so that they can only be used in the
+expected, correct way. 
+```go
+// Bad: do all `cache` users, current and future know how to use it correctly?  
+var (
+	cache = NewCache()
+)
+...
+func (w *widget) Update(x X) error {
+    w.mutex.Lock()
+	defer w.mutex.Unlock()
+	
+	if cache.Contains(x) {
+	    ...	
+    }
+	...
+}
+```
+
+```go
+// Better: `cache` users likely understand that manipulations require
+// synchronization might miss future changes to this contract. 
+var (
+	cache = NewCache()
+    mutex sync.RWMutex
+)
+...
+func (w *widget) Update(x X) error {
+    mutex.Lock()
+	defer mutex.Unlock()
+	
+	if cache.Contains(x) {
+	    ...	
+    }
+	...
+}
+```
+
+```go
+// Good: `cache` is only used by `widget` methods likely living in the same file.
+type widget struct {
+  cache Cache
+  mutex sync.RWMutex
+}
+...
+func (w *widget) Update(x X) error {
+    w.mutex.Lock()
+	defer w.mutex.Unlock()
+	
+	if w.cache.Contains(x) {
+	    ...	
+    }
+	...
+}
+```
+
+Capture in comments assumptions and decisions that are not obvious from the
+code, for example:
+```go
+var (
+	authorizer = perrpc.FromMap(map[authz.Authorizer][]string{
+		// When this endpoint was public, Sensor relied on it to check Central's
+		// availability. While Sensor might not do so today, we need to ensure
+		// backward compatibility with older Sensors.
+		or.SensorOr(user.Authenticated()): {
+			"/v1.MetadataService/GetMetadata",
+		},
+	})
+)
+```
+
+## Guidelines
+
+- Always use `defer mutex.Unlock()` instead of explicitly calling `Unlock()`.
+- A `Close()` on a file etc. should only be deferred and wrapped in
+  `utils.IgnoreError` if you exclusively _read_. If you write to a file, you
+  **must** check for a `nil` error upon close, otherwise you cannot be certain
+  that all data has been persisted.
+- Use the fact that certain operations return zero values by default to write
+  conditions in a more goal-oriented way. E.g., prefer
+  `if x := m[k]; x != nil { foo(*x) }` over `if x, ok := m[k]; ok { foo(*x) }`.
+  The former makes it apparent that the `*x` cannot panic while the latter only
+  after checking that `m` indeed can never contain `nil` values.
+
+### Declarations
```

Comment:
I'm not sure whether we want to extend this to also other things, but if we were I'd like to add the usage of
```go
var (
   _ Interface = (*structImplementingInterface)(nil)
)
```

as a way of documenting within the `var` block which interfaces the struct implements.

Reply:
Called this out but not a var-block requirement.